### PR TITLE
3 - Aducm support for iio_demo

### PR DESCRIPTION
**Pull request by mchindri**
Build on top of: #657 
Just last 2 commit are relevant. The others are from the other pull request.

### CODE REVIEW

**Comment by mchindri** [comment]
v2: Fix codacy

File: libraries/iio/iio.c [hunk 1]
```diff
@@ -0,0 +1,843 @@
+/***************************************************************************//**
```
Comment (cristiantpop) [inline libraries/iio/iio.c:1]: Isn't this copy realized in a previous pull request?
https://github.com/analogdevicesinc/no-OS/pull/657/commits/d30c3e72833be2882dc9a2d70a41b7ce30b7c544

File: iio/iio_demo/iio_demo.c [hunk 2]
```diff
@@ -53,6 +53,13 @@
 /********************** Macros and Constants Definitions **********************/
 /******************************************************************************/
 
+#ifdef ADUCM_PLATFORM
+
+#define MAX_BUFF_SIZE 5000
+static uint8_t g_buff[MAX_BUFF_SIZE];
```
Comment (adrimbarean) [inline iio/iio_demo/iio_demo.c:59]: nitpick: this could also be a buffer passed from main. This would have the benefit that it keeps the initialization structure identical across platforms. Would it be worth to do it this way?

File: projects/iio_demo/Makefile [hunk 3]
```diff
@@ -1,7 +1,23 @@
...
+
+SRC_DIRS += projects/$(TARGET)/src
+SRC_DIRS += iio/iio_app
+SRC_DIRS += iio/iio_demo
+
+ifeq (y,$(strip $(TINYIIOD)))
+SRC_DIRS += libraries/iio
+endif
+
+include ../../tools/scripts/aducm.mk
```
Comment (adrimbarean) [inline projects/iio_demo/Makefile:13]: It seems to me like this change should not happen here. The existing makefiles compile for every platform with respect to the operating system in which the compilation is done. The new makefile aducm.mk compiles for the aducm platform with respect to the operating system. Should support for the aducm platform be added to the windows and linux makefiles instead?

Comment (adrimbarean) [reply projects/iio_demo/Makefile:13]: That way, this file does not change anymore, I believe.

File: projects/iio_demo/src/main.c [hunk 4]
```diff
@@ -192,7 +229,9 @@ int main(void)
 	iio_demo_out_init_par = (struct iio_demo_init_param) {
 		.name = demo_device_output,
 		.num_channels = 4,
+#ifdef XILINX_PLATFORM
 		.ddr_base_addr = DAC_DDR_BASEADDR,
```
Comment (cristiantpop) [inline projects/iio_demo/src/main.c:233]: `ddr_base_addr` could be initialized with the address of the `static uint8_t g_buff[MAX_BUFF_SIZE];`, declared in main.c

File: projects/iio_demo/Makefile (hunk 3 above)
Comment (mchindri) [reply projects/iio_demo/Makefile:13]: I think we should create a generic makefile for both windows and linux that do what this makefile dose or create a generic makefile slited in multiple file with specific functionalities.
But I think this is not related to this PR

File: iio/iio_demo/iio_demo.c (hunk 2 above)
Comment (mchindri) [reply iio/iio_demo/iio_demo.c]: moved to main

File: projects/iio_demo/src/main.c (hunk 4 above)
Comment (mchindri) [reply projects/iio_demo/src/main.c]: done

**Comment by mchindri** [comment]
v3:
 - Rebase on other changes
 - move g_buff to main as in_buff and out_buff and use them as ddr_base_addr

**Comment by mchindri** [comment]
v4:
 - Rebase onto master
 - Split first commit in 2 commits
 - Remove makefile related commit. Will add an other PR for it

**Comment by mchindri** [comment]
will merge today if no more comments

File: iio/iio_demo/iio_demo.c [hunk 5]
```diff
@@ -297,10 +297,12 @@ static ssize_t iio_demo_write_dev(void *iio_inst, char *buf,
 	demo_device = (struct iio_demo_device *)iio_inst;
```
Comment (mirceacaprioru) [inline iio/iio_demo/iio_demo.c:297]: typo in commit description iio_demo

File: projects/iio_demo/src/main.c [hunk 6]
```diff
@@ -192,7 +244,7 @@ int main(void)
 	iio_demo_out_init_par = (struct iio_demo_init_param) {
 		.name = demo_device_output,
 		.num_channels = 4,
-		.ddr_base_addr = DAC_DDR_BASEADDR,
+		.ddr_base_addr = OUTPUT_BASE_ADDRESS,
```
Comment (mirceacaprioru) [inline projects/iio_demo/src/main.c:247]: if you make this change the demo won't work with xilinx anymore. Did you test this ?

Comment (mirceacaprioru) [reply projects/iio_demo/src/main.c:247]: Forget this i just saw the define above. Maybe just use one define for Xilinx also not use another define for an existing one just change the name to OUTPUT_BASE_ADDRESS. Also try to separate the modifications in more commits it is hard to follow this change in just one commit named "Aducm support for iio_demo".

File: projects/iio_demo/pinmux_config.c [hunk 7]
```diff
@@ -0,0 +1,40 @@
+/*
```
Comment (mirceacaprioru) [inline projects/iio_demo/pinmux_config.c:1]: commit description is longer than 75 characters/line.

File: iio/iio_demo/iio_demo.c (hunk 5 above)
Comment (mchindri) [reply iio/iio_demo/iio_demo.c:297]: done

File: projects/iio_demo/src/main.c (hunk 6 above)
Comment (mchindri) [reply projects/iio_demo/src/main.c]: done

File: projects/iio_demo/pinmux_config.c (hunk 7 above)
Comment (mchindri) [reply projects/iio_demo/pinmux_config.c:1]: done

**Comment by mchindri** [comment]
v5:
 - Split iio_demo aducm support in multiple commits
 - Fix typo
 - Split commit message on 75 character lines

File: projects/iio_demo/src/main.c [hunk 8]
```diff
@@ -157,9 +198,17 @@ int main(void)
 	};
 #endif // XILINX_PLATFORM
 
+#ifdef ADUCM_PLATFORM
+
+	/* Aducm platform dependent initialization for UART. */
+	struct aducm_uart_init_param platform_uart_init_par = {
+			.parity = UART_NO_PARITY,
+			.stop_bits = UART_ONE_STOPBIT,
+			.word_length = UART_WORDLEN_8BITS };
```
Comment (adrimbarean) [inline projects/iio_demo/src/main.c:207]: nitpick: end bracket should be on new line.

File: projects/iio_demo/src/parameters.h [hunk 9]
```diff
@@ -43,12 +43,21 @@
...
+#ifdef ADUCM_PLATFORM
+#include "irq_extra.h"
+#endif
 
 /******************************************************************************/
 /********************** Macros and Constants Definitions **********************/
 /******************************************************************************/
 
+
+#ifdef XILINX_PLATFORM
```
Comment (adrimbarean) [inline projects/iio_demo/src/parameters.h:59]: I am not sure we need an `#ifdef XILINX_PLATFORM` here if we already check the `_XPARAMETERS_PS_H_` define. `_XPARAMETERS_PS_H_` should exist only on Xilinx PS platforms.

Comment (mchindri) [reply projects/iio_demo/src/parameters.h:59]: I use XILINX_PLATFORM because it is defined in app_config.h

**Comment by mchindri** [comment]
v6:
 - Fix astyle
 - Move braket on new line

**Comment by mchindri** [comment]
Will merge today if no more comments

File: projects/iio_demo/src/main.c (hunk 8 above)
Comment (mchindri) [reply projects/iio_demo/src/main.c]: done
